shell/profiling/profiler.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/profiling/profiler.h"
+
+namespace flutter {
+
+Profiler::Profiler(fml::RefPtr<fml::TaskRunner> profiler_task_runner,
+                   CpuUsage cpu_usage,
```

Comment:
What's the plan when we add more to the profiler, such as memory or gpu usages?

We can add `GpuUsage`, `MemoryUsage`, ..., to the initialization list but it can get very long.

Shall we just send in a `Sampler`  instead of `CpuUsage` (or `CpuUsageGenerator`)? Each platform can implement its own `Sampler` where calling `Sampler()` returns a `ProfileSample` that currently only has cpu usages, but can later expand?

Reply:
Yup, makes sense.